discovery

```diff
@@ -0,0 +1,216 @@
+#!/bin/sh
+set -e
+
+[ -n "$WEAVE_DEBUG" ] && set -x
+
+usage() {
+cat <<-EOF
```

Comment:
This is a much nicer way of outputting usage -- the weave script ought to use it too :)